.github/workflows/docs.yml

```diff
@@ -0,0 +1,74 @@
+# Workflow to build and publish the documentation
+name: Documentation builder
+
+on:
+  schedule:
+    # Nightly build at 3:42 A.M.
+    - cron: "42 3 */1 * *"
+  pull_request:
+    branches:
+      - main
+      - "[0-9]+.[0-9]+.X"
```

Comment:
We can not really do this for pull requests, because we do not want to push `scikit-learn/scikit-learn.github.io` for pull requests.

There are a lot of moving parts here. I propose the following set PRs to get this going:

1. Get docs to build on github actions for nightly and release branches and store them as artifacts. We are still need to keep circleci for hosting PRs. This can be done by refactoring `build_tools/circle/build_doc.sh` to not depend on CI specific environment variables or a total new script for github actions.
2. Extend 1, by actually deploying to `scikit-learn/scikit-learn.github.io` and remove  `deploy` in circleci.
3. See if the idea in #21068 works for coordinating between github actions and circleci so that we can reduce the load on circleci.

Reply:
So, the idea is to build the documentation with `actions` but deploy with `circleci` so that we can render the documentation, is it not?

Reply:
There are two distinct way to "deploy":

- full doc deployments to `scikit-learn/scikit-learn.github.io` (nightly for `main` that deploys to `https://scikit-learn.org/dev`, manual commits for release branches that deploy to `https://scikit-learn.org/X.Y`)
- temporary deployments as browseable circleci artifacts for pull-request (to check the sphinx HTML rendering before merging a PR to main). This is what is used to populate the following CI doc status in PRs: 

<img width="819" alt="image" src="https://user-images.githubusercontent.com/89061/135286175-00879084-90eb-49ab-a489-f75964c27671.png">

For the latter, @thomasjpfan suggested a plan in #21068 to build the doc in github actions and publish the result via a lightweight and fast circleci ci job that would graph the result of the github actions doc build.

Unfortunately it's not possible to directly publish browseable HTML artifacts directly in the as a result of github actions run, see: https://github.com/scikit-learn/scikit-learn/issues/21068#issuecomment-927704912.